File: Data Structures & Algorithms/trapping-rain-water/submission-6.java
class Solution {
    public int trap(int[] height) {
        int left=height[0],right=height[height.length-1],res=0;
        for(int l=0,r=height.length-1;l<r;){
            if(left<=right){
                l++;
                left=Math.max(left,height[l]);
                res+=left-height[l];
            }
            else{
                r--;
                right=Math.max(right,height[r]);
                res+=right-height[r];
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/lowest-common-ancestor-in-binary-search-tree/submission-6.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def lowestCommonAncestor(self, root: TreeNode, p: TreeNode, q: TreeNode) -> TreeNode:
        stack=[root]
        res=None
        while stack:
            curr=stack.pop()
            if not curr:
                break
            if p.val<curr.val and q.val<curr.val:
                stack.append(curr.left)
            elif p.val>curr.val and q.val>curr.val:
                stack.append(curr.right)
            else:
                res=curr
                break
        return res